test/integration/backward_compatible/parallel/serialization/compact/CompactSerializersLiveTest.js

```diff
@@ -112,9 +136,16 @@ describe('CompactSerializersLiveTest', function () {
                 await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
                 const map = await client.getMap(mapName);
                 const value = await map.get(1);
-                value.should.be.instanceof(CompactUtil.EmployeeDTO);
-                value.age.should.be.equal(expectedAge);
-                value.id.equals(expectedId).should.be.true;
+                // Test result can be changed by configuration
+                if (name == 'Zero config') {
+                    value.should.be.instanceof(CompactUtil.EmployeeDTO);
+                    value.age.should.be.equal(expectedAge);
+                    value.id.equals(expectedId).should.be.true;
+                } else {
+                    value.schema.typeName.should.be.equal('employee');
+                    value.values['age'].should.be.equal(expectedAge);
+                    value.values['id'].equals(expectedId).should.be.true;
```

Comment:
For typename there is no Public API to retrieve it so it is fine. However, can you use public APIs to retrieve the fields? Using public APIs in tests makes them more resilient to internal changes.

Reply:
Fixed.